src/types/version.go

```diff
@@ -39,6 +41,13 @@ type Docker struct {
 	Privileged     bool           `json:"privileged,omitempty"`
 }
 
+func (d *Docker) Valid() error {
```

Comment:
I don't this is the good way to valid version as 1) version struct under type package are intend to be imported by other project; 2) will make docker validation into multiple occurrences.